django/db/models/fields/tuple_lookups.py

```diff
@@ -304,7 +306,12 @@ def as_sqlite(self, compiler, connection):
         return root.as_sql(compiler, connection)
 
     def as_subquery(self, compiler, connection):
-        return compiler.compile(In(self.lhs, self.rhs))
+        lhs = self.lhs
+        rhs = self.rhs
+        if isinstance(lhs, ColPairs):
+            rhs.set_values([source.name for source in lhs.sources])
```

Comment:
A copy of `rhs` should be created here to avoid causing hard to identify side effects if the expression is reused for resolving or compilation.

```suggestion
            rhs = rhs.clone()
            rhs.set_values([source.name for source in lhs.sources])
```

Reply:
Thank you for catching that!